app/frontend/wallet/helpers/request.js

```diff
@@ -19,6 +19,13 @@ const request = async function request(url, method = 'GET', body = null, headers
     if (response.status === 429) {
       await sleep(DELAY_AFTER_TOO_MANY_REQUESTS)
       return await request(url, method, body, headers)
+    } else if (response.status >= 500) {
```

Comment:
this would get out of the request function as a generic "NetworkError" because it's re-caught on line 36. I covered that case in the refactor I suggested in the comment, of course, feel free to improve it

Reply:
Also, I would remove the try/catch blocks throwing the generic "NetworkError" in cardano-wallet.js, otherwise if those calls fail with a "ServerError" it would not propagate to the "message-rendering" layer and the user would see a misleading error message.

Reply:
I see - wouldn't that debuglog be handled by the global error handler https://github.com/vacuumlabs/adalite/blob/5edd51631c1fa1e93d3bbe96b5ba4a82d4f17f1a/app/frontend/actions.ts#L74 though?

Reply:
hm, :D yeah it would, thanks, :+1: fixing...